### PR TITLE
Bump API version and clean up changelog.json

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:
This PR bumps the API version to v2.23 (the next release) and cleans up the `changelog.json` file, which contains 2.22 changes.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind cleanup

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: modules/web/src/assets/config/changelog.json [hunk 1]
```diff
@@ -1,193 +1,4 @@
 {
   "externalChangelogURL": "https://github.com/kubermatic/kubermatic/blob/main/docs/changelogs/CHANGELOG-2.22.md",
```
Comment (Waseem826) [inline modules/web/src/assets/config/changelog.json:2]: Should we also update version in this URL?

Comment (embik) [reply modules/web/src/assets/config/changelog.json:2]: There is not going to be a 2.23 changelog for quite some time. Do you think it's okay to have a 404 url?

Comment (Waseem826) [reply modules/web/src/assets/config/changelog.json:2]: ah, if its going to land on 404 then its fine to keep it like this.

**Review by Waseem826 — APPROVED** [review]

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**